src/Cli/dotnet/commands/dotnet-test/Program.cs

```diff
@@ -32,7 +32,7 @@ public static int Run(ParseResult parseResult)
             // from the VSTest side.
             string testSessionCorrelationId = $"{Environment.ProcessId}_{Guid.NewGuid()}";
 
-            string[] args = parseResult.GetArguments();
+            string[] args = parseResult.GetArguments().Concat(parseResult.UnmatchedTokens).ToArray();
```

Comment:
As we discussed in chat this adds duplicate parameters to the args that are passed directly to VSTest. 

Instead we need fixing in the path that is a bit lower that sends the parameters to msbuild. 

  .WithEnvironmentVariable("DOTNET_CLI_VSTEST_TRACE", "1") writes how we call msbuild to screen, and can aid in debugging this. 
  
There seems to be special handling for CollectCoverage, and it also differs when a test project is provided or not. 

Will have another look tomorow.